spec/views/layout/application.html.erb_spec.rb

```diff
@@ -0,0 +1,116 @@
+require 'rails_helper'
+
+RSpec.describe 'navigation bar', type: :view do
+
+  shared_examples 'a registered User' do
+    it 'should link to home' do
+      expect(page).to have_link('Wimi Portal', href: current_url)
+    end
+
+    it 'should link to profile' do
+      expect(page).to have_link('Profile', href: user_path(@user))
+    end
+
+    it 'should link to logout' do
+      expect(page).to have_link('Logout', href: destroy_user_session_path)
+    end
+  end
+
+  context 'for a superadmin' do
+    it_behaves_like 'a registered User'
+    before(:each) do
+      @user = FactoryGirl.create(:user, superadmin: true)
+      login_as @user
+      visit root_path
+    end
+
+    it 'should show 3 links' do
+      expect(page).to have_css('nav a', count: 3)
+    end
+  end
+
+  context 'for a wimi' do
+    it_behaves_like 'a registered User'
+    before(:each) do
+      @chair = FactoryGirl.create(:chair)
+      @wimi_user = FactoryGirl.create(:user)
+      @user = FactoryGirl.create(:wimi, user: @wimi_user, chair: @chair).user
+      login_as @user
+      visit root_path
+    end
+
+    it 'should link to projects' do
+      expect(page).to have_link('Projects', href: projects_path)
+    end
+
+    it 'should show 4 links' do
+      expect(page).to have_css('nav a', count: 4)
+    end
+  end
+
+  context 'for a hiwi' do
+    it_behaves_like 'a registered User'
+    before(:each) do
+      @user = FactoryGirl.create(:user)
+      @chair = FactoryGirl.create(:chair)
+      @project = FactoryGirl.create(:project, chair: @chair, status: true)
+      @user.projects << @project
+      login_as @user
+      visit root_path
+    end
+
+    it 'should link to projects' do
+      expect(page).to have_link('Projects', href: projects_path)
+    end
+
+    it 'should show 4 links' do
+      expect(page).to have_css('nav a', count: 4)
```

Comment:
@tbjoern Wie gesagt, es geht halt nicht nur darum, dass die Links vorhanden sind, sondern _genau_ diese Links dort sind. Dieser Test besagt in dem Kontext halt, dass der normale Nutzer zB nicht noch einen Link zu "Chairs" sieht oder ähnliches. Damit ich aber nicht explizit auf das Nicht-Vorhandensein von jedem möglichen Link prüfen muss, sage ich einfach, dass die Gesamtanzahl die der bisher beschriebenen Links nicht überschreiten darf.
Oder siehst du das anders und ich verstehe nur deinen Kommentar falsch?